src/spatialdata/_core/centroids.py

```diff
@@ -0,0 +1,146 @@
+
+
+def _get_centroids_for_axis(xdata: xr.DataArray, axis: str) -> pd.DataFrame:
+    """
+    Compute the component "axis" of the centroid of each label as a weighted average of the xarray coordinates.
+
+    Parameters
+    ----------
+    xdata
+        The xarray DataArray containing the labels.
+    axis
+        The axis for which the centroids are computed.
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame containing one column, named after "axis", with the centroids of the labels along that axis.
+        The index of the DataFrame is the collection of label values, sorted ascendingly.
+    """
+    centroids: dict[int, float] = defaultdict(float)
+    for i in xdata[axis]:
+        portion = xdata.sel(**{axis: i}).data
+        u = da.unique(portion, return_counts=True)
+        labels_values = u[0].compute()
+        counts = u[1].compute()
+        for j in range(len(labels_values)):
+            label_value = labels_values[j]
+            count = counts[j]
+            centroids[label_value] += count * i.values.item()
+
+    all_labels_values, all_labels_counts = da.unique(xdata.data, return_counts=True)
+    all_labels = dict(zip(all_labels_values.compute(), all_labels_counts.compute()))
+    for label_value in centroids:
+        centroids[label_value] /= all_labels[label_value]
+    centroids = dict(sorted(centroids.items(), key=lambda x: x[0]))
+    return pd.DataFrame({axis: centroids.values()}, index=list(centroids.keys()))
+
+
+@get_centroids.register(SpatialImage)
+@get_centroids.register(MultiscaleSpatialImage)
+def _(
+    e: SpatialImage | MultiscaleSpatialImage,
+    coordinate_system: str = "global",
+) -> DaskDataFrame:
+    model = get_model(e)
+    if model in [Image2DModel, Image3DModel]:
+        raise ValueError("Cannot compute centroids for images.")
+    assert model in [Labels2DModel, Labels3DModel]
+    _validate_coordinate_system(e, coordinate_system)
+
+    if isinstance(e, MultiscaleSpatialImage):
+        assert len(e["scale0"]) == 1
+        e = SpatialImage(next(iter(e["scale0"].values())))
+
+    dfs = []
+    for axis in get_axes_names(e):
+        dfs.append(_get_centroids_for_axis(e, axis))
+    df = pd.concat(dfs, axis=1)
+    t = get_transformation(e, coordinate_system)
+    centroids = PointsModel.parse(df, transformations={coordinate_system: t})
+    return transform(centroids, to_coordinate_system=coordinate_system)
+
+
+@get_centroids.register(GeoDataFrame)
+def _(e: GeoDataFrame, coordinate_system: str = "global") -> DaskDataFrame:
+    _validate_coordinate_system(e, coordinate_system)
+    t = get_transformation(e, coordinate_system)
+    assert isinstance(t, BaseTransformation)
+    # separate points from (multi-)polygons
+    first_geometry = e["geometry"].iloc[0]
+    if isinstance(first_geometry, Point):
+        xy = e.geometry.get_coordinates().values
+    else:
+        assert isinstance(first_geometry, (Polygon, MultiPolygon))
```

Comment:
Can you please add a message to this `assert` so people know what the error is for?